Moodlet/app/src/main/java/com/cmput3owo1/moodlet/services/MoodEventService.java

```diff
@@ -123,6 +127,34 @@ public void editMoodEvent(MoodEvent moodEvent, OnMoodUpdateListener listener){
 
     }
 
+    /**
+     * Delete swiped MoodEvent from database.
+     * @param context The activity information.
+     * @param moodEvent The mood event to be deleted.
+     * @param listener The listener to notify upon completion of deletion.
+     */
+    @Override
+    public void deleteMoodEvent(final Context context, MoodEvent moodEvent, final OnMoodUpdateListener listener) {
+        DocumentReference newMoodEventRef = db.collection("moodEvents").document(moodEvent.getId());
+
+
+        newMoodEventRef.delete()
+                .addOnSuccessListener(new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void aVoid) {
+                        listener.onMoodUpdateSuccess();
+                    }
+                })
+
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Toast.makeText(context, R.string.delete_failed, Toast.LENGTH_SHORT).show();
```

Comment:
Yeah like James said, you can move this out of here by adding a "onMoodUpdateFailure" function to the OnMoodUpdateListener interface. Then you don't need to pass in "context"

Reply:
Oops I realized James just said to add the toast message for failure, didn't mention moving it out. But yeah it might be better to move it out so you don't have to pass in the activity context. In the onFailure function you can just call listener.onMoodDeleteFailure

Reply:
Oops, that's what I meant by it adding a Toast too but should've explained, TYTY